octobot_trading/modes/script_keywords/basic_keywords/account_balance.py

```diff
@@ -0,0 +1,103 @@
+#
+#  You should have received a copy of the GNU Lesser General Public
+#  License along with this library.
+
+import octobot_commons.constants as commons_constants
+import octobot_commons.symbols.symbol_util as symbol_util
+import octobot_trading.constants as trading_constants
+import octobot_trading.enums as trading_enums
+import octobot_trading.errors as trading_errors
+import octobot_trading.personal_data as trading_personal_data
+
+
+async def total_account_balance(context, side=trading_enums.PositionSide.BOTH.value):
+    current_price = \
+        await trading_personal_data.get_up_to_date_price(context.exchange_manager,
+                                                         symbol=context.symbol,
+                                                         timeout=trading_constants.ORDER_DATA_FETCHING_TIMEOUT)
+    on_inverse_contract = False
+    if context.exchange_manager.is_future:
+        # for BTC/USD
+        # on linear, return balance in currency (ex: BTC)
+        # on inverse, return balance in market (ex: USD)
+        on_inverse_contract = context.exchange_manager.exchange_personal_data.positions_manager.get_symbol_position(
+            context.symbol, side
+        ).symbol_contract.is_inverse_contract()
+    value = context.exchange_manager.exchange_personal_data.\
+        portfolio_manager.portfolio_value_holder.portfolio_current_value
+    base, quote = symbol_util.parse_symbol(context.symbol).base_and_quote()
+    reference_market = context.exchange_manager.exchange_personal_data.portfolio_manager.reference_market
+    if reference_market == quote:
+        return value if on_inverse_contract else value / current_price
+    if reference_market == base:
+        return value * current_price if on_inverse_contract else value
+    raise NotImplementedError(f"Impossible to compute total account balance for a symbol that doesn't contain "
+                              f"the reference market. reference market: {reference_market}, symbol: {context.symbol}")
+
+
+def _get_locked_amount_in_stop_orders(context, side):
+    locked_amount = trading_constants.ZERO
+    for order in context.exchange_manager.exchange_personal_data.orders_manager.get_open_orders(context.symbol):
+        if isinstance(order, (trading_personal_data.StopLossOrder, trading_personal_data.StopLossLimitOrder)) and \
+           order.side.value == side:
+            locked_amount += order.origin_quantity
+    return locked_amount
+
+
+async def available_account_balance(context, side=trading_enums.TradeOrderSide.BUY.value, use_total_holding=False,
+                                    is_stop_order=False, reduce_only=True):
+    portfolio_type = commons_constants.PORTFOLIO_TOTAL if use_total_holding else commons_constants.PORTFOLIO_AVAILABLE
+    current_symbol_holding, current_market_holding, market_quantity, current_price, symbol_market = \
+        await trading_personal_data.get_pre_order_data(context.exchange_manager,
```

Comment:
It can take time if the price is not available, it this case it's just waiting. Otherwise it's very fast, almost nothing is done, I think we can call it as much as we want. the async part is the to allow waiting for the price if missing

Reply:
Ok great 👍